### DISCUSSION TITLE
function that tells if given date is in this week or Not.

### DISCUSSION THREAD

**Discussion by Akhilbisht798** [js-fns/date-fns#3205] [General] [answered]
hey, I am currently working with ```date-fns``` and there was some problem that I have to check if the given date is in this week or not.

I have checked  ```date-fns```  documentation and can't seem to find any methods that can help with this.

I am not sure if there is any function or not. 
But if there is none it would be great to have this function for week and/or month.
```Just a suggestion.```

**Comment by Pooja444** [comment]
@Akhilbisht798 I would like to take up this issue
Can this issue be assigned to me? Also, will it count towards Hacktoberfest 2022 contributions?
@fturmel

**Comment by Akhilbisht798** [comment]
@Pooja444 Hey, I am currently new to this project and don't have much experience working on it. I opened this new issue as a suggestion for the people who are more experienced than me in this.
If they suggest that to start working on it then I don't mind you working on it.

**Comment by fturmel** [comment, marked as answer]
@Akhilbisht798 I don't think there's a function to specifically handle this use case, but you can easily get there with `startOfWeek`:

```js
import { startOfWeek } from "date-fns";

function isDateInCurrentWeek(date) {
  return startOfWeek(new Date()).getTime() === startOfWeek(date).getTime();
}

console.log(isDateInCurrentWeek(new Date(2022, 9, 4))); // true
console.log(isDateInCurrentWeek(new Date(2022, 9, 2))); // true
console.log(isDateInCurrentWeek(new Date(2022, 9, 1))); // false
```

@Pooja444 I'm not on the core team, so I can't confirm but there was a recent addition to the README that hints that it will. https://github.com/date-fns/date-fns/commit/9ecc9b266a22d6ca89debc913b6e5c0a9b9d7ab6

>  **Warning**: the current `main` represents v3 pre-release version of the library. [See `v2` branch](https://github.com/date-fns/date-fns/tree/v2).
> If you're participating in hacktoberfest, please send your PRs to `main`.

**Comment by Akhilbisht798** [comment]
@fturmel hey, your solution is great for this problem. 

Do I keep this issue open for further discussion on this idea or just close it?

**Comment by fturmel** [comment]
It's simple enough that I don't think it's adding a lot of value, but since the library is tree-shakable there's no file size penalty for adding more helper functions. You could always work on a PR, but I have no idea if it might get accepted and merged.

I'll move this to discussions for now so it'll be easier for the community to find the proposed solution.

**Comment by Akhilbisht798** [comment]
Oh, it will we great to make it into the discussion.
